### ISSUE TITLE
The results of masked.log_softmax on MPS are inconsistent with those on CPU

### ISSUE THREAD

**Issue by CaoE** [pytorch/pytorch#111416]
### 🐛 Describe the bug

The output of `masked.log_softmax` on CPU with inputs of cpu_args:  [tensor(-8.4844, dtype=torch.float16, requires_grad=True), 0] and cpu_kwargs:  {'mask': tensor(False)},  i.e., the masked element, is nan. While the result on MPS is 0. https://github.com/pytorch/pytorch/actions/runs/6542102776/job/17764726988.
`masked.softmax` and `masked.softmin` should have the same issue.

### Versions

PyTorch: main branch.

cc @jbschlosser @bhosmer @cpuhrsch @erichan1 @drisspg @mikaylagawarecki @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev @ezyang @gchanan @zou3519 @kadeng

**Comment by jbschlosser** [comment]
I believe this is the "entire row masked out" case. Although inconsistent and arguably incorrect, returning 0 instead of NaN does avoid problems like those reported in e.g. #41508.

**Comment by skotapati** [comment]
On current MacOS 14.6.1 with the latest torch nightly, mps behaves consistently with cpu & this issue no longer repros
